### PR TITLE
feat(TimePicker): promote component to core

### PR DESCRIPTION
**Pull request by plagoa**

### CODE REVIEW

**Review by francisco-guilherme — CHANGES_REQUESTED** [review]
You forgot to add the component both to index.js and index.d.ts.

File: packages/core/src/TimePicker/TimePicker.js [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+  const minutesInputState =
+    time?.minutes != null && time.minutes !== "" && time.minutes >= 0 && time.minutes <= 59;
+  const secondsInputState =
+    time?.seconds != null && time.seconds !== "" && time.seconds >= 0 && time.seconds <= 59;
+
+  return hourInputState && minutesInputState && secondsInputState;
+};
+
+/**
+ * A TimePicker component used to choose the time, following specifications provided by Design System. Still in development.
```
Comment (francisco-guilherme) [inline packages/core/src/TimePicker/TimePicker.js:62]: Please remove the "Still in development".
Maybe the "following specifications provided by Design System" is redundant.
It seems good enough just "A TimePicker component used to choose the time."

Comment (plagoa) [reply packages/core/src/TimePicker/TimePicker.js:62]: Well spotted, I'll remove it.

**Review by francisco-guilherme — APPROVED** [review]

File: packages/core/src/TimePicker/enums.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const TimePickerUnits = {
```
Comment (eliofreitas) [inline packages/core/src/TimePicker/enums.js:1]: all the objects in this enum file could use an Object.freeze() to prevent any undesired modification

**Comment by plagoa** [comment]
@eliofreitas addressed your issues.

**Review by eliofreitas — APPROVED** [review]

**Pull request merged**